### PR TITLE
Zappy 8 architecture

### PR DESCRIPTION
**Pull request by Kapsulon**

### CODE REVIEW

File: server/include/game.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+** EPITECH PROJECT, 2024
+** Zappy
+** File description:
+** game
+*/
+
+#ifndef GAME_H_
+    #define GAME_H_
+
+#include <stdlib.h>
```
Comment (lmaoleo) [inline server/include/game.h]: Coding style error : Tab

**Review by lmaoleo — CHANGES_REQUESTED** [review]
game.h : Coding Style Error

File: server/include/connection.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    int fd;
+    struct sockaddr_in addr;
+    char *buffer;
+    char *team;
+    struct connection_s *next;
+} connection_t;
+
+connection_t *add_connection(connection_t *cons, connection_t *new);
+void free_connection(connection_t *cons);
+connection_t *new_connection(int fd, struct sockaddr_in addr, char *team);
```
Comment (Klaiveft) [inline server/include/connection.h]: Use pointer for struct MAJOR C-F7

File: server/src/connection/new_connection.c [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/*
+** EPITECH PROJECT, 2024
+** Zappy
+** File description:
+** new_connection
+*/
+
+#include "connection.h"
+
+connection_t *new_connection(int fd, struct sockaddr_in addr, char *team)
```
Comment (Klaiveft) [inline server/src/connection/new_connection.c]: use Pointer for struct in definitions of functions MAJOR C-F7

File: server/src/main.c [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+** Zappy
+** File description:
+** main
+*/
+
+#include "server.h"
+
+int main(int argc, char **argv)
+{
+    server_t *server = NULL;
```
Comment (Klaiveft) [inline server/src/main.c:12]: separate initialisation form the function MINOR C-L6

**Review by Klaiveft — CHANGES_REQUESTED** [review]

**Review by PaulArbez — APPROVED** [review]

**Review by Klaiveft — APPROVED** [review]
SGTM

**Review by lmaoleo — APPROVED** [review]
👍

**Review by albe2 — APPROVED** [review]
good

**Pull request merged**